src/signals/incident/containers/IncidentContainer/reducer.js

```diff
@@ -78,7 +82,81 @@ export default (state = initialState, action) => {
           ...action.payload,
         })
       )
+    case ADD_TO_SELECTION: {
+      const selected = action.payload[action.payload.meta_name]?.selection[0]
+      const previousSelection = state.get('incident').toJS()[
+        action.payload.meta_name
+      ]?.selection
+      let selection = [selected]
+      if (
+        selected.type !== NEARBY_TYPE &&
+        previousSelection &&
+        previousSelection[0].type !== NEARBY_TYPE
+      ) {
+        selection = [
+          ...(state
+            .get('incident')
+            .toJS()
+            [action.payload.meta_name]?.selection.filter(
+              ({ id }) =>
+                id !== action.payload[action.payload.meta_name]?.selection[0].id
+            ) || []),
+          selected,
+        ]
+      }
+
+      return state.set(
+        'incident',
+        fromJS({
+          ...state.get('incident').toJS(),
+          ...{
+            [action.payload.meta_name]: {
+              selection,
+              location: action.payload.location,
+            },
+          },
+        })
+      )
+    }
+    case REMOVE_FROM_SELECTION: {
+      const updated = state
+        .get('incident')
+        .toJS()
+        [action.payload.meta_name]?.selection.filter(
+          ({ id }) =>
+            id !== action.payload[action.payload.meta_name]?.selection[0].id
+        )
 
+      return state.set(
+        'incident',
+        fromJS({
+          ...state.get('incident').toJS(),
+          ...{
+            [action.payload.meta_name]: {
+              selection: updated.length > 0 ? updated : undefined,
+              location: {
+                address: updated[0]?.address,
+                coordinates: updated[0]?.coordinates,
+              },
+            },
+          },
+        })
+      )
+    }
+    case REMOVE_SELECTION: {
```

Comment:
op dit niveau kan die ook weg, je moet dan wel deze logica zetten op de plek waar je checked of 
 ```
action.payload[action.payload.meta_name]?.selection[0] 
```
leeg is als het goed is

Reply:
done